src/slice.rs

```diff
@@ -83,24 +96,23 @@ pub(crate) struct SliceHeader {
 /// Struct to hold all the actual payload of a [`Slice`].
 ///
 /// This is what actually gets "shredded" into different [`Shred`]s.
+#[derive(Serialize, Deserialize)]
 pub(crate) struct SlicePayload {
-    pub(crate) data: Vec<u8>,
+    parent: Option<(Slot, Hash)>,
+    data: Vec<u8>,
 }
 
-impl SlicePayload {
-    /// Constructs a new [`SlicePayload`] from the given `data`.
-    pub(crate) fn new(data: Vec<u8>) -> Self {
-        Self { data }
-    }
-
-    /// Returns the size of the payload in bytes.
-    pub(crate) fn len(&self) -> usize {
-        let SlicePayload { data } = self;
-        data.len()
+impl From<SlicePayload> for Vec<u8> {
+    fn from(payload: SlicePayload) -> Self {
+        bincode::serde::encode_to_vec(payload, bincode::config::standard()).unwrap()
```

Comment:
bit unfortunate that we need to allocate an entirely new `Vec` here, would be great if we could reuse the memory from `data`, but probably not possible with `bincode`